File: app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel, HttpUrl

from app.schemas.youtube import YouTubeVideoMetadata


class AnalyzeCommentsRequest(BaseModel):
    youtube_url: HttpUrl


class SentimentBreakdown(BaseModel):
    positive: int = 0
    neutral: int = 0
    negative: int = 0


class AnalyzeCommentResult(BaseModel):
    youtube_comment_id: str
    author: str | None = None
    text_original: str
    sentiment: str
    confidence: float
    predicted_at: datetime
    published_at: datetime | None = None


class AnalyzeCommentsResponse(BaseModel):
    video_id: str
    video_url: str
    video: YouTubeVideoMetadata
    sentiment_breakdown: SentimentBreakdown
    total_comments: int
    total_predictions: int
    predictions: list[AnalyzeCommentResult]

File: web/src/types/analysis.ts
export interface YouTubeVideoMetadata {
  title: string;
  thumbnail_url: string;
  view_count: number;
  like_count: number;
  comment_count_total: number | null;
}

export interface SentimentBreakdown {
  positive: number;
  neutral: number;
  negative: number;
}

export interface AnalyzeCommentResult {
  youtube_comment_id: string;
  author: string | null;
  text_original: string;
  sentiment: string;
  confidence: number;
  predicted_at: string;
  published_at: string | null;
}

export interface AnalyzeCommentsResponse {
  video_id: string;
  video_url: string;
  video: YouTubeVideoMetadata;
  sentiment_breakdown: SentimentBreakdown;
  total_comments: number;
  total_predictions: number;
  predictions: AnalyzeCommentResult[];
}
